### PR TITLE
[bug 1206861] PB Logo Swap (2of2)

### PR DESCRIPTION
**Pull request by ckprice**
This is snippet 2 of 3 for Private Browsing. See bug for full specs.

Stage: https://snippets.allizom.org/admin/base/snippet/96/

@glogiotatidis r?

### CODE REVIEW

**Comment by glogiotatidis** [comment]
I see expected launch date 11 Nov so I'll push reviewing those to next week due to bedrock work.

File: campaigns/private-browsing/logo-swap.html [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+              brandLogo.style.opacity = 1;
+          };
+          for (var k = 0, l = xbuttons.length; k < l; k++) {
+              var xbutton = xbuttons[k];
+              xbutton.addEventListener('mouseover', mouseover, false);
+              xbutton.addEventListener('mouseout', mouseout, false);
+          }
+        {% endif %}
+
+        var buttons = snippet.getElementsByClassName('launch-pb');
```
Comment (glogiotatidis) [inline campaigns/private-browsing/logo-swap.html]: Please use `snippet_id`

File: campaigns/private-browsing/logo-swap.html [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+        isTargetAvailable('privateWindow', function targetAvailable () {
+          sendMetric('impression-pb-available');
+          for (var i=0; i < buttons.length; i++) {
```
Comment (glogiotatidis) [inline campaigns/private-browsing/logo-swap.html]: Similar microoptimization as in https://github.com/mozilla/snippets/pull/56/files#r42121205

File: campaigns/private-browsing/logo-swap.html [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+<div id="snippet-{{ snippet_id }}" class="snippet">
+  {% if blockable %}
+    <button class="block-snippet-button block-snippet-button-{{ snippet_id }}">
+    </button>
+  {% endif %}
+  {% if blockable or clickable %}
+    <div id="block-snippet-overlay-{{ snippet_id }}" class="block-snippet-overlay-{{ snippet_id }}">
+    </div>
+  {% endif %}
+  {% if fallback_url %}<a href="{{ fallback_url|safe }}" class="launch-pb">{% endif %}
```
Comment (glogiotatidis) [inline campaigns/private-browsing/logo-swap.html]: Fallback_url is required in this snippet, correct?

**Comment by glogiotatidis** [comment]
* Logo replacement works :+1: 
* Clickable logo works :+1: 
* Snippet blocking works :+1: 
* Clickable snippet works :+1:
* Fallback urls work :+1: 

nits with `snippet_id`, loops, fallback_url

**Comment by ckprice** [comment]
@glogiotatidis updates made, thanks!

**Comment by glogiotatidis** [comment]
Merged with a few optimizaitons here caebf1fc4ffc4de5ef2c83e7f1b034a6366b1f53

Thanks @ckprice

**Pull request merged**